### ISSUE TITLE
Placing a unit in the scenario editor crashes wesnoth

### ISSUE THREAD

**Issue by Pentarctagon** [wesnoth/wesnoth#7296]
Reported on the forums for 1.16.7 (https://forums.wesnoth.org/viewtopic.php?t=56614) - placing a unit (I placed a Vampire Bat) on the map via the scenario editor causes Wesnoth to crash with an assertion.

@stevecotton I haven't bisected it, but I would guess that it's caused by https://github.com/wesnoth/wesnoth/commit/4d1fc268b17ba6166cda8ba52a956c8675e11073

Backtrace:
```
unit::get_ability_bool(const unit * const this, const std::string & tag_name, const map_location & loc) (/home/derpy/Desktop/wesnoth/wesnoth-dev/src/units/abilities.cpp:159)
unit::invisible(const unit * const this, const map_location & loc, bool see_all) (/home/derpy/Desktop/wesnoth/wesnoth-dev/src/units/unit.cpp:2432)
unit_drawer::redraw_unit(const unit_drawer * const this, const unit & u) (/home/derpy/Desktop/wesnoth/wesnoth-dev/src/units/drawer.cpp:146)
display::draw_invalidated(display * const this) (/home/derpy/Desktop/wesnoth/wesnoth-dev/src/display.cpp:2583)
display::draw(display * const this, bool update, bool force) (/home/derpy/Desktop/wesnoth/wesnoth-dev/src/display.cpp:2522)
display::draw(display * const this) (/home/derpy/Desktop/wesnoth/wesnoth-dev/src/display.cpp:2479)
events::raise_draw_event() (/home/derpy/Desktop/wesnoth/wesnoth-dev/src/events.cpp:782)
controller_base::play_slice(controller_base * const this, bool is_delay_enabled) (/home/derpy/Desktop/wesnoth/wesnoth-dev/src/controller_base.cpp:392)
editor::editor_controller::main_loop(editor::editor_controller * const this) (/home/derpy/Desktop/wesnoth/wesnoth-dev/src/editor/controller/editor_controller.cpp:178)
editor::start(const std::string & filename, bool take_screenshot, const std::string & screenshot_filename) (/home/derpy/Desktop/wesnoth/wesnoth-dev/src/editor/editor_main.cpp:65)
game_launcher::start_editor(game_launcher * const this, const std::string & filename) (/home/derpy/Desktop/wesnoth/wesnoth-dev/src/game_launcher.cpp:1044)
game_launcher::start_editor(game_launcher * const this) (/home/derpy/Desktop/wesnoth/wesnoth-dev/src/game_launcher.hpp:113)
do_gameloop(const std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > & args) (/home/derpy/Desktop/wesnoth/wesnoth-dev/src/wesnoth.cpp:943)
main(int argc, char ** argv) (/home/derpy/Desktop/wesnoth/wesnoth-dev/src/wesnoth.cpp:1113)
```

**Comment by stevecotton** [comment]
> I would guess that it's caused by https://github.com/wesnoth/wesnoth/commit/4d1fc268b17ba6166cda8ba52a956c8675e11073

Argh, yes looks likely.

**Comment by knyghtmare** [comment]
Also, confirmed on 1.17.12+dev

**Comment by stevecotton** [comment]
This greatly improves the situation, but there's still a crash if an ability wants to check which units are adjacent. For example:

* Open the editor
* Create a new scenario
* Place a Shadow
* Place another unit next to the Shadow

Happens with Shadows (regardless of time of day) and Elvish Rangers (regardless of terrain).

My feeling is that this is still a big improvement, and it's a good trade off between known issues vs chance of introducing a new bug.

**Comment by stevecotton** [comment]
Fixed for 1.16 in #7297. The issue with Nightstalk (and other `[hides]` abilities) was fixed in the version that merged.
